package kr.co.jboard.dao;

import java.util.ArrayList;
import java.util.List;

import kr.co.jboard.dto.ArticleDTO;
import kr.co.jboard.utill.DBHelper;
import kr.co.jboard.utill.SQL;

public class CommentDAO extends DBHelper {
	
	// 싱글톤
	private static CommentDAO instance = new CommentDAO();
	public static CommentDAO getInstance() {
		return instance;
	}
	private CommentDAO() {}
	
	// 기본 CRUD 메서드
	public ArticleDTO select(String ano) {
		
		ArticleDTO dto = null;
		
		try {
			conn = getConnection();
			psmt = conn.prepareStatement(SQL.SELECT_ARTICLE);
			psmt.setString(1, ano);
			
			psmt.executeQuery();
			if(rs.next()) {
				dto = new ArticleDTO();
				dto.setAno(rs.getInt(1));
				dto.setType(rs.getString(2));
				dto.setTitle(rs.getString(3));
				dto.setContent(rs.getString(4));
				dto.setComment(rs.getInt(5));
				dto.setFile(rs.getInt(6));
				dto.setHit(rs.getInt(7));
				dto.setWriter(rs.getString(8));
				dto.setRegip(rs.getString(9));
				dto.setWdate(rs.getString(10));
			}
			closeAll();
			
			
		} catch(Exception e) {
			e.printStackTrace();
		}
		return dto;
	}
	public List<ArticleDTO> selectAll() {
		
		// 반환용 List
		List<ArticleDTO> dtoList = new ArrayList<>();
		
		try {
			conn = getConnection();
			stmt = conn.createStatement();
			
			rs = stmt.executeQuery(SQL.SELECT_ALL_ARTICLE);
			while(rs.next()) {
				ArticleDTO dto = new ArticleDTO();
				dto = new ArticleDTO();
				dto.setAno(rs.getInt(1));
				dto.setType(rs.getString(2));
				dto.setTitle(rs.getString(3));
				dto.setContent(rs.getString(4));
				dto.setComment(rs.getInt(5));
				dto.setFile(rs.getInt(6));
				dto.setHit(rs.getInt(7));
				dto.setWriter(rs.getString(8));
				dto.setRegip(rs.getString(9));
				dto.setWdate(rs.getString(10));
				
				dtoList.add(dto);
			}
			
			closeAll();
			
		} catch(Exception e) {
			e.printStackTrace();
		}
		
		
		return dtoList;
	}
	public void insert(ArticleDTO dto) {
		try {
			conn = getConnection();
			psmt = conn.prepareStatement(SQL.INSERT_ARTICLE);
			psmt.setString(1, dto.getTitle());
			psmt.setString(2, dto.getContent());
			psmt.setString(3, dto.getWriter());
			psmt.setString(4, dto.getRegip());
			psmt.executeUpdate();
			
			closeAll();
			
			
			
		} catch(Exception e) {
			e.printStackTrace();
		}
		
	}
	public void update(ArticleDTO dto) {
		
		try {
			conn = getConnection();
			psmt = conn.prepareStatement(SQL.UPDATE_ARTICLE);
			psmt.setString(1, dto.getTitle());
			psmt.setString(2, dto.getContent());
			psmt.setInt(3, dto.getAno());
			psmt.executeUpdate();
			
			closeAll();
			
			
			
		} catch(Exception e) {
			e.printStackTrace();
		}
		
	}
	public void delete(String ano) {
		try {
			conn = getConnection();
			psmt = conn.prepareStatement(SQL.DELETE_ARTICLE);
			psmt.setString(1, ano);
			psmt.executeUpdate();
			
			closeAll();
			
			
			
		} catch(Exception e) {
			e.printStackTrace();
		}
	}
}
